src/index.ts

```diff
@@ -59,7 +59,7 @@ export default async function GleeAppInitializer() {
   await registerAuth(GLEE_AUTH_DIR)
 
   const parsedAsyncAPI = await getParsedAsyncAPI()
-  const channelNames = parsedAsyncAPI.channelNames()
+  const channelNames = parsedAsyncAPI.channels().map(e => e.address())
```

Comment:
I have mixed feelings about this 😅 I see what you're trying to do here but I think it's gonna bite you hard down the track because you're assuming that there can't be multiple channels with the same address, which is not true.

If I were you, I'd leave the array of `channelNames` as the array of channel names and would use it to retrieve the address whenever I need it.

Reply:
I created a new property in the `Adapter class` called `channelAddresses` which returns the address and changed `channelNames` to return the `Id`'s  and when I need the URL I am doing something like `this.parsedAsyncAPI.channels().get(channelName).address()` this way we don't have to touch the logging logic for spec v3.